src/relay/ranker/mod.rs

```diff
@@ -0,0 +1,167 @@
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex, MutexGuard};
```

Comment:
We run with tokio framework, we need to user Mutex from tokio which is implemented for coroutines. This mutex lock method will lock the entire thread. Using the mutex from tokio has an async lock method. This means that when u locking the mutex, if it got blocked it only pause the current routing not the entire thread hence the framework can schedule other routines until the lock is available.

Reply:
This means that this entire module need to be implemented in async-io otherwise it will not play nice with the rest of the system

Reply:
switched to RWLock from tokio